app/Http/Controllers/OAuthController.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Laravel\Socialite\Facades\Socialite;
+use App\Models\User;
+use Illuminate\Support\Facades\Auth;
+
+class OAuthController extends Controller
+{
+    public function redirect($socialPlatform) {
+        return Socialite::driver($socialPlatform)->redirect();
+    }
+
+    public function loginUsingSocial($socialPlatform) {
+        $socialUser = Socialite::driver($socialPlatform)->user();
```

Comment:
show error message if don't have provider_id response

Reply:
Noted, sir.

Reply:
By show message, do you mean show message in the blade/view or just in console sir?

Reply:
error message in the .blade